lucene/misc/src/java/org/apache/lucene/misc/store/DirectIODirectory.java

```diff
@@ -51,16 +50,9 @@
  * to the provided Directory instance.
  *
  * <p>See <a
- * href="{@docRoot}/overview-summary.html#NativeUnixDirectory">Overview</a>
+ * href="{@docRoot}/overview-summary.html#DirectIODirectory">Overview</a>
  * for more details.
  *
- * <p>To use this you must compile
- * NativePosixUtil.cpp (exposes Linux-specific APIs through
- * JNI) for your platform, by running <code>./gradlew build</code>, and then putting the resulting
- * <code>libLuceneNativeIO.so</code> or <code>libLuceneNativeIO.dylib</code>
- * (from <code>lucene/misc/native/build/lib/release/platform/</code>) onto your dynamic
- * linker search path.
- *
  * <p><b>WARNING</b>: this code is very new and quite easily
  * could contain horrible bugs.  For example, here's one
  * known issue: if you use seek in <code>IndexOutput</code>, and then
```

Comment:
Ah good catch! I updated this as well as surrounding docs to also include Window compatibility. 

One thing I wasn't sure though and feel like asking is, shall I also remove this line :D 

>  * <p><b>WARNING</b>: this code is very new and quite easily could contain horrible bugs. 

As it's been around for some time now (although not used by many yet?), and being pure-java means it's less error prone.

What do you think?

Reply:
Let's leave the warning for now?  Because this was JNI code, bringing it risks like SIGSEGV / memory leaks / etc. (even though in this instances those risks were very small), and requiring C++ compilation to build per-platform, very few users tried it out.

Now that it will be pure java (thank you @zacharymorn!!), it should see more usage.

Especially for search applications that do concurrent indexing and searching on a single box (e.g. Elasticsearch!), this Directory should be a huge win to bring down interference of a large merge running on a box and impacting queries as hot pages are swapped out.

So, let's wait and see if we can remove this warning at a later date, if users really do use this and have good results?

Also, this directory requires some tuning (the buffer size).

Reply:
Sounds good that makes sense. Thanks for the detailed explanation Mike!